stream_alert/rule_processor/parsers.py

```diff
@@ -219,16 +219,19 @@ def _parse_records(self, schema, json_payload):
             return [json_payload]
 
         envelope_schema = self.options.get('envelope_keys')
-        # If envelope_keys are declared, and this payload does not have every key specified
-        # in these envelope_keys, then it's safe to skip trying to extract records using json_path
-        if envelope_schema:
+        optional_envelope_keys = self.options.get('optional_envelope_keys')
+
+        if envelope_schema and optional_envelope_keys:
```

Comment:
Small nit - can you just add a code comment explaining the two different behaviors?